### ISSUE TITLE
Implement Cookie/Consent Management?

### ISSUE THREAD

**Issue by aterjung** [OpenMage/magento-lts#1030]
What do you think about implementing an own Cookie/Consent Management into OpenMage?

A few years ago, in this thread it was part of a more general GDPR discoussion that was closed without result:

https://github.com/OpenMage/magento-lts/issues/362

As the EOL is very near by - would it be good to start this discussion again? At lest an integrated Consent Management would be helpfull to me. Every project has totaly different implementations of Cookie banners and switches, often with third party js snippets calling a third partys server...
I would like to have alle of these integrated in OpenMage with Backend-Tools to configure and document the consent. What do you think about that?

**Comment by Flyingmana** [comment]
as someone who added this already to shops, its a heavy legal topic which can move very strongly in short timeframes. I dont think we have the resources to offer a reliable solution as part of the core.

But, we could offer the space to develop it as an independent module.

my personal recommendation is to use one of the existing 3th party modules, despite them usually being paid modules. At least for now.

**Comment by colinmollenhour** [comment]
I agree that these days a modern cookie/consent management feature does belong as an out of the box feature in a shopping cart. However, I also generally agree with @Flyingmana that there could be a good deal of legal research both initially and ongoing to build and maintain the feature. As it can easily be tacked on using an extension I don't think it harms OpenMage to not have this functionality built-in. Even if you or someone else is interested in building and donating it to the OpenMage core (which I'm sure we'd greatly appreciate) there is the risk that if the original author abandons it there is one more feature for the core to maintain. 

So while it would be nice to see the features list of OpenMage expand, there are probably going to be a significant number of features that we prefer as extensions so that the OpenMage team can focus on core stability, security and performance and the broader community can continue to contribute new features by way of extensions.

There is a "Suggested Modules" page on the website and I would love to see that expanded, like an "Awesome OpenMage" page. So far it has languished so I'm not sure what to make of it. With Magento Connect shutting down it would be nice for there to be an OpenMage-hosted alternative but so far it seems nobody has time to put into it.

**Comment by sreichel** [comment]
Create un "feature" project for relateted tickets and better overview?

**Comment by aterjung** [comment]
what i have in my mind is not so big like the modules from amasty & co. Thinking of these, yes, i agree totaly. But it would be absolut ok to develop this as an "OpenMage Module" - probably there are parts that would would better fit into the core - but that can be discussed when the implementation is there. I have to rework a project on this topic in the next week, probably a good start for a module development. Would you supply a Repository?
